source/gui/_addonStoreGui/viewModels/store.py

```diff
@@ -264,22 +245,75 @@ def helpAddon(self, listItemVM: AddonListItemVM) -> None:
 		assert path is not None
 		startfile(path)
 
-	def removeAddon(self, listItemVM: AddonListItemVM[_AddonGUIModel]) -> None:
+	def removeAddon(
+			self,
+			listItemVM: AddonListItemVM[_AddonGUIModel],
+			askConfirmation: bool = True,
+			useRememberChoiceCheckbox: bool = False,
+	) -> tuple[bool, bool]:
+		from gui import mainFrame
 		assert addonDataManager
 		assert listItemVM.model
-		if _shouldProceedToRemoveAddonDialog(listItemVM.model):
+		if askConfirmation:
+			res, shouldRememberChoice = _shouldProceedToRemoveAddonDialog(
+				mainFrame,
+				listItemVM.model,
+				useRememberChoiceCheckbox=useRememberChoiceCheckbox,
+			)
+		else:
+			res = True
+			shouldRememberChoice = True
+		if res:
 			addonDataManager._deleteCacheInstalledAddon(listItemVM.model.name)
 			assert listItemVM.model._addonHandlerModel is not None
 			listItemVM.model._addonHandlerModel.requestRemove()
 			self.refresh()

```

Comment:
should refreshed be moved here too like in the changes in `installOverrideIncompatibilityForAddon`

Reply:
This is unanswered

```suggestion
		self.refresh()
```

Reply:
> should refreshed be moved here too like in the changes in `installOverrideIncompatibilityForAddon`

No, that's the other way around, i.e. `installOverrideIncompatibilityForAddon` to be fixed. There is no need to refresh if no action has occurred.